portfolio/src/main/webapp/script.js

```diff
@@ -68,28 +77,30 @@ const RIGHT_KEY = 39;
 const UP_KEY = 38;
 const DOWN_KEY = 40;
 
-//move snake inside the canvas using arrow keys
+const STEP_SNAKE = 20;
+
+// move snake inside the canvas using arrow keys
 window.onkeydown = function(event) {
   for (var z = 4; z > 0; z--) {
     x[z] = x[(z - 1)];
     y[z] = y[(z - 1)];
   }
+
+  var values = getCanvasAndContext();
+  var canvas = values.canvas;
   
-  var keyPr = event.keyCode; //Key code of key pressed
+  var keyPr = event.keyCode; // Key code of key pressed
   
-  if(keyPr == RIGHT_KEY && x[0]<=canvas.width - 40) { 
-    x[0] = x[0]+20;
-  }
-  else if(keyPr == LEFT_KEY && x[0]>10) {
-    x[0] = x[0]-20; 
-  }
-  else if(keyPr == UP_KEY && y[0]>10) {
-    y[0] = y[0]-20; 
-  }
-  else if(keyPr == DOWN_KEY && y[0]<=canvas.height - 40) {
-    y[0] = y[0]+20; 
+  if(keyPr == RIGHT_KEY && x[0] <= canvas.width - 40) { 
+    x[0] = x[0] + STEP_SNAKE;
+  } else if(keyPr == LEFT_KEY && x[0] > 10) {
+    x[0] = x[0] - STEP_SNAKE; 
+  } else if(keyPr == UP_KEY && y[0] > 10) {
+    y[0] = y[0] - STEP_SNAKE; 
+  } else if(keyPr == DOWN_KEY && y[0] <= canvas.height - 40) {
+    y[0] = y[0] + STEP_SNAKE; 
```

Comment:
if the expected range is 10 to width/height - 40, after going into the if block, the new value will have the range as (-20, width/height - 10], is this your expected range?

Reply:
The expected range after going into the if block is (-10, width/height - 20), since STEP_SNAKE = 20.

Reply:
I by accident thought it was 30, anyway it is better you use step_snake in the condition to make the expected range more readable and more importantly to avoid unexpected range when you change the step_snake value, something like (-10 + step_length, width/height - 20 - step_length]

btw, is negative value expected?

Reply:
Yes, the negative value is expected for the left boundary because of the formatting of the canvas and the section container that contains it. I might change the formatting later so that the range becomes STEP_SNAKE to width/height - STEP_SNAKE.